docs/spec/relayer-v2/RelayerDefinitions.tla

```diff
@@ -0,0 +1,282 @@
+ChannelOrder == {"ORDERED", "UNORDERED"}
+
+Max(S) == CHOOSE x \in S: \A y \in S: y <= x
+
+(******************************* ChannelEnds *******************************
+    A set of channel end records. 
+    A channel end record contains the following fields:
+    
+    - state -- a string 
+      Stores the current state of this channel end. It has one of the 
+      following values: "UNINIT", "INIT", "TRYOPEN", "OPEN", "CLOSED".
+      
+    - channelID -- a channel identifier
+      Stores the channel identifier of this channel end.  
+    
+    - counterpartyChannelID -- a channel identifier
+      Stores the channel identifier of the counterparty channel end.
+      
+      (** ignoring channel ordering for now **) 
+ ***************************************************************************)   
+ChannelEnds ==
+    [
+        state : ChannelStates, 
+        channelID : ChannelIDs \union {nullChannelID},
+        counterpartyChannelID : ChannelIDs \union {nullChannelID}
+    ] 
+
+(***************************** ConnectionEnds *****************************
+    A set of connection end records. 
+    A connection end record contains the following fields:
+    
+    - state -- a string 
+      Stores the current state of this connection end. It has one of the 
+      following values: "UNINIT", "INIT", "TRYOPEN", "OPEN".
+    
+    - connectionID -- a connection identifier
+      Stores the connection identifier of this connection end.
+    
+    - counterpartyConnectionID -- a connection identifier
+      Stores the connection identifier of the counterparty connection end.
+    
+    - clientID -- a client identifier
+      Stores the client identifier associated with this connection end. 
+      
+    - counterpartyClientID -- a client identifier
+      Stores the counterparty client identifier associated with this connection end.
+ ***************************************************************************)
+ConnectionEnds ==
+    [
+        state : ConnectionStates,
+        connectionID : ConnectionIDs \union {nullConnectionID},
+        clientID : ClientIDs \union {nullClientID},
+        counterpartyConnectionID : ConnectionIDs \union {nullConnectionID},
+        counterpartyClientID : ClientIDs \union {nullClientID}, 
+        channelEnd : ChannelEnds
+    ]  
+
+
+(********************************** Chains *********************************
+    A set of chain records. 
+    A chain record contains the following fields:
+    
+    - height : an integer between nullHeight and MaxHeight. 
+      Stores the current height of the chain.
+    
+    - counterpartyClientHeights : a set of integers between 1 and MaxHeight
+      Stores the heights of the client for the counterparty chain.
+
+    - connectionEnd : a connection end record 
+      Stores data about the connection with the counterparty chain
+ ***************************************************************************)
+Chains ==    
+    [
+        height : Nat,
```

Comment:
Why not 1..MaxHeight?

Reply:
This module contains definitions of operators which are reused across other modules (they extend `RelayerDefinitions.tla`, and thus get access to the definitions). `MaxHeight` is a constant of the environment, and its value cannot be transferred to an extended module. This definition of `Chains` is only used in the type invariant, to check if a given chain store is a record from this set. TLC does not use the definition of `Chains` to compute states, hence it does not complain. The chain stores restrict height to be in the set `1..MaxHeight`, so the type invariant ensures that the height (between `1..MaxHeight`) is a natural number.